### PR TITLE
Updated BlockDetectorAllLocomotiveTrack to make it more cooler.

### PR DESCRIPTION
**Pull request by FloofyPeachy**
Here is what I updated. I've updated the Locomotive Detector track so it can be changed with a crowbar.

1: **_WHACK_** it with a crowbar. You can set it to steam, electric, diesel, or all trains.

A bit of testing may be required to make sure everything works. (I did already)

### CODE REVIEW

**Comment by EternalBlueFlame** [comment]
you know, you never did upload the fixed version of this that we talked about.....

File: src/main/java/train/common/blocks/tracks/BlockDetectorAllLocomotiveTrack.java [hunk 1]
```diff
@@ -1,26 +1,92 @@
...
+			return false;
+		}
+		ItemStack current = player.inventory.getCurrentItem();
+		if ((current != null) && ((current.getItem() instanceof IToolCrowbar))) {
+			IToolCrowbar crowbar = (IToolCrowbar) current.getItem();
+			if (crowbar.canWhack(player, current, getX(), getY(), getZ())) {
+					this.ThingToSet += 3;
+				switch(this.ThingToSet) {
+					case 0:{
+  player.addChatMessage(new ChatComponentText("Now set to emit a signal on all electric trains"));
```
Comment (NitroxydeX) [inline src/main/java/train/common/blocks/tracks/BlockDetectorAllLocomotiveTrack.java:40]: "Detector All Locomotives" and the first Case is "electric trains"?

File: src/main/java/train/common/blocks/tracks/BlockDetectorAllLocomotiveTrack.java [hunk 2]
```diff
@@ -1,26 +1,92 @@
...
+	}
+	
 	@Override
 	public void onMinecartPass(EntityMinecart cart) {
-		if (cart instanceof Locomotive) {
-			setTrackPowering();
+		
+		switch(this.ThingToSet) {
+					case 0:{
+setTrackPowering();
```
Comment (NitroxydeX) [inline src/main/java/train/common/blocks/tracks/BlockDetectorAllLocomotiveTrack.java:72]: No check if the EntityMinecart is instanceof the desired locomotive

File: src/main/java/train/common/blocks/tracks/BlockDetectorAllLocomotiveTrack.java [hunk 3]
```diff
@@ -1,26 +1,92 @@
...
+					case 6: {
+						player.addChatMessage(new ChatComponentText("Now set to emit a signal on all steam trains"));
+						break;
+					}
+					case 9:{
+							player.addChatMessage(new ChatComponentText("Now set to emit a signal on all trains"));
+					break;
+					}
+				}
+				this.ThingToSet += 3;
```
Comment (NitroxydeX) [inline src/main/java/train/common/blocks/tracks/BlockDetectorAllLocomotiveTrack.java:57]: you already set this in line 37

**Pull request merged**